src/models/builders/nextjs-builder.ts

```diff
@@ -1,190 +1,71 @@
-                        try {
-                            const contents = await readFile(functionConfigFile, "utf8");
-                            functionConfig = JSON.parse(contents);
-                        } catch {
-                            this.invalidFunctions.push(file);
-                        }
-
-                        if (functionConfig.runtime !== "edge") {
-                            this.invalidFunctions.push(name);
-                        }
-
-                        const functionFile = join(filepath, functionConfig.entrypoint);
-                        let functionFileExists = false;
-                        try {
-                            await stat(functionFile);
-                            functionFileExists = true;
-                        } catch {
-                            console.log("Error in stat function file")
-                        }
-
-                        if (!functionFileExists) {
-                            this.invalidFunctions.push(name);
-                        }
-
-                        let contents = await readFile(functionFile, "utf8");
-                        contents = contents.replace(
-                            /Object.defineProperty\(globalThis,\s*"__import_unsupported",\s*{[^}]*}\)/gm,
-                            "true"
-                        );
-
-                        // minify removed !
-
-                        const newFilePath = join(this.tmpFunctionsDir, `${relativePath}.js`);
-                        await mkdir(dirname(newFilePath), { recursive: true });
-                        await writeFile(newFilePath, contents);
-
-                        this.functionsMap.set(
-                            relative(functionsDir, filepath).slice(0, -".func".length),
-                            newFilePath
-                        );
-                    } else if (isDirectory) {
-                        await this.dirWalk(filepath, functionsDir);
-                    }
-                })
-            );
-        } catch (error) {
-            const message = `Error: ${error}`;
-            console.log(message)
-            throw Error(message)
-        }
-    }
-
-    async handleMiddleware() {
+    handleMiddleware() {
         console.log("Handling middleware ...");
 
         try {
             let middlewareManifest: any = {};
 
             middlewareManifest = JSON.parse(
-                await readFile(".next/server/middleware-manifest.json", "utf8")
+                readFileSync(".next/server/middleware-manifest.json","utf8")
             );
 
+            if(!middlewareManifest.middleware || !middlewareManifest.functions){
+                throw new MiddlewareManifestHandlerError('Missing properties in middleware-manifest.json');
+            }
+
             this.middlewareEntries = Object.values(middlewareManifest.middleware);
             this.functionsEntries = Object.values(middlewareManifest.functions);
 
+            if(this.functionsMap.size <= 0){
```

Comment:
I suggest to move this if to line 161. If we don't have any function on the map we don't need to call handleMiddleware.

Reply:
done